### PR TITLE
feat: load data from url

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes https://github.com/zakodium-oss/analysis-ui-components/issues/219

### CODE REVIEW

**Comment by lpatiny** [comment]
In the meantime we have removed filelist and we need now to create a fileCollection using https://github.com/cheminfo/filelist-utils/blob/aa6402dd26b3592a6073c52ed85bffa1904478d2/src/fileCollectionFromWebservice.ts#L19-L20

The parameter in the URL should also be replaced. We could use fileCollectionURL as parameter name for example. 

I have already updated:

https://zakodium-oss.github.io/analysis-dataset/full.json

And now the JSON use 'relativePath' instead of 'webkitRelativePath' that requires the new version of filelist-utils and the use of fileCollectionFromWebservice.

**Comment by wadjih-bencheikh18** [comment]
test link :
https://deploy-preview-240--analysis-ui-components.netlify.app/?filelist=https://zakodium-oss.github.io/analysis-dataset/full.json

**Comment by wadjih-bencheikh18** [comment]
> And now the JSON use 'relativePath' instead of 'webkitRelativePath' that requires the new version of filelist-utils and the use of fileCollectionFromWebservice.

https://github.com/cheminfo/biologic-converter is used in the project too, so we need to update it first

**Comment by stropitek** [comment]
@wadjih-bencheikh18 can you resolve the conflicts?

File: src/app/App.tsx [hunk 1]
```diff
@@ -51,7 +53,11 @@ function DropZoneArea() {
   const dispatch = useAppDispatch();
   const appState = useAppState();
   const measurement = getCurrentMeasurement(appState);
-
+  const url = useMemo(() => {
+    const paramsString = window.location.href.split('?')[1];
+    const searchParams = new URLSearchParams(paramsString);
```
[the remark this answers is not in the document]
Comment (lpatiny) [reply src/app/App.tsx:58]: @wadjih-bencheikh18 Please also take care that we are allowed before the question mark to put '#'. This is what we use in NMRium

https://www.nmrium.org/nmrium#?nmrium=https://cheminfo.github.io/nmr-dataset-demo/coffee/coffee.nmrium

**Comment by lpatiny** [comment]
@wadjih-bencheikh18 Please first of all rebase the project to the main branch. We did update filelist and now we are speaking about 'FileCollection'.

File: src/app/App.tsx [hunk 2]
```diff
@@ -63,6 +69,15 @@ function DropZoneArea() {
...
   ];
+  async function getData() {
+    if (url) {
+      const filelist = await fileListFromWebservice(url);
+      await loadFiles(filelist, dispatch);
+    }
+  }
+  useEffect(() => {
+    void getData();
+  });
```
[the remark this answers is not in the document]
Comment (wadjih-bencheikh18) [reply src/app/App.tsx]: fixed

File: src/app/App.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (targos) [reply src/app/App.tsx]: @stropitek why not `new URLSearchParams(window.location.search)` ?

[the remark this answers is not in the document]
Comment (stropitek) [reply src/app/App.tsx]: Yes, even better, forgot about `window.location.search`

**Comment by stropitek** [comment]
The example with the link is not working. I think the data is downloaded but the application cannot parse the data

image

**Comment by lpatiny** [comment]
This error is likely to be related to biologic converter. Like me check this tonight.

**Comment by lpatiny** [comment]
@wadjih-bencheikh18 Please could you rebase the project to test ?

I also did a simplified version of the data that contains only .jdx files

https://deploy-preview-240--analysis-ui-components.netlify.app/?filelist=https://zakodium-oss.github.io/analysis-dataset/jdx.json

And this one works correctly so @stropitek for me it can be merged.

**Review by lpatiny — APPROVED** [review]

**Pull request merged**